src/App.js

```diff
@@ -1,17 +1,40 @@
-import React,{useState} from 'react'
+import React,{useState ,useEffect} from 'react'
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
+
+function useSemiPersistentState () {
+  const [todoList, setTodoList]= useState (
+    localStorage.getItem ('savedTodoList') ? JSON.parse (localStorage.getItem ('savedTodoList')): [] 
+    )
+  useEffect(()=> { localStorage.setItem('savedTodoList',JSON.stringify(todoList));
+
+},[todoList])
+
+
+return[todoList, setTodoList];
```

Comment:
return statement should be matching useEffect and useState that are under the same outer branch in useSemiPersistentState